### ISSUE TITLE
Agda unable to compile base library

### ISSUE THREAD

**Issue by kellino** [NixOS/nixpkgs#28466]
## Issue description
I've just started using nixos, and so far everything is working perfectly apart from Agda. I see that previous issues, #15617 and #15306 concerned compilation of the stdlib and agdaPrelude, but I cannot get Agda to compile anything at all. Even for a simple file, not dependent on any libraries, I get the following error message:

``
/nix/store/26hl2dmpk05m5bgybvp42686rfmcgzl8-Agda-2.5.2-data/share/ghc-8.0.2/x86_64-linux-ghc-8.0.2/Agda-2.5.2/lib/prim/Agda/Primitive.agdai:
openBinaryFile: permission denied (Read-only file system)
``

Looking in this directory tells me that Agda has not been able to compile its base libraries as it is unable to write its *.agdai files in the directory. I've no idea how to solve this. I've tried both installing via my configuration.nix and also using nix-env. Neither seems to make a difference. I have GHC installed only via nix-env.

### Steps to reproduce
nix-env -f "\<nixpkgs\>" -iA haskellPackages.Agda 
agda emptyFile.agda

## Technical details
* System: 17.09pre112899.129f8d7e99 (Hummingbird)
* Nix version: nix-env (Nix) 1.11.13
* Nixpkgs version: "17.09pre112899.129f8d7e99"
* Sandboxing enabled: true

Any help would be much appreciated!

**Comment by m0rphism** [comment]
I can confirm the problem.

As a temporary workaround I've used the Agda package from an older nixpkgs revision by adding a package override to my `~/.nixpkgs/config.nix`:

    {
      packageOverrides = pkgs :
        let
          fetchNixPkgs = { rev, sha256 }:
            pkgs.fetchFromGitHub {
              inherit sha256 rev;
              owner = "NixOS";
              repo = "nixpkgs-channels";
            };
          pinnedPkgs = import (fetchNixPkgs {
            rev = "2180d2c1180b04b14877cccad841fdb06941255a";
            sha256 = "0kf5babd9y7r5wz4982m637x65lh8m4qma6gpc9mix1bxp2bvh8q";
          }) {};
        in {
          pinned = pinnedPkgs;
        };
    }

and installed it via

    nix-env -i -A nixos.pkgs.pinned.haskellPackages.Agda

**Comment by kellino** [comment]
Thanks @m0rphism, that worked a treat. I doubt I would have ever come up with this solution myself. To install the standard library will I have to do the same thing and use an older version of nixpkgs?

**Comment by m0rphism** [comment]
Correct. Installing `pinned.AgdaStdlib` and making sure emacs runs the Agda executable with `--include-dir=/home/user/.nix-profile/share/agda` worked for me.

**Comment by kellino** [comment]
Again, thanks to @m0rphism, installing the libraries is also easy via this method. I'm going to leave this issue open though as it is a workaround rather than a solution.
